APE17.rst

```diff
@@ -0,0 +1,482 @@
+  experimental version of `extension-helpers
+  <https://github.com/astrofrog/extension-helpers>`_.
+* `This SunPy branch <https://github.com/sunpy/sunpy/compare/master...Cadair:no_more_helpers>`_
+  shows how that package can use astropy_helpers only as a build time dependency
+  for building a C extension
+* The following pure-Python packages have dropped astropy-helpers and adopted
+  some or all of the recommendations outlined here:
+
+  * DKIST (`3552bbeb <https://github.com/DKISTDC/dkist/commit/3552bbeb12af0e58ae48fee447d2222f0619ec03>`_)
+  * MOSViz (`spacetelescope/mosviz#179 <https://github.com/spacetelescope/mosviz/pull/179>`_)
+  * CubeViz (`spacetelescope/cubeviz#492 <https://github.com/spacetelescope/cubeviz/pull/492>`_).
+  * aas-timeseries (`aperiosoftware/aas-timeseries#492 <https://github.com/aperiosoftware/aas-timeseries/pull/29>`_)
+
+Implementation
+--------------
+
+All the changes to packages are already described above, but to summarize, the
+following table shows the correspondence between old astropy-helpers features
+and their proposed replacement:
+
++-------------------------------------+-------------------------------------+
+| Feature                             | Replacement                         |
++-------------------------------------+-------------------------------------+
+| ``astropy.version_helpers``         | `setuptools_scm`_                   |
++-------------------------------------+-------------------------------------+
+| Package data in specification       | All package data should be          |
+| in ``setup_package.py`` files       | specified in ``setup.cfg``          |
++-------------------------------------+-------------------------------------+
+| ``python setup.py test``            | `tox`_ or direct use of `pytest`_   |
++-------------------------------------+-------------------------------------+
+| ``python setup.py build_docs``      | `tox`_ or ``make html``             |
+|                                     | or ``sphinx-build``                 |
++-------------------------------------+-------------------------------------+
+| Delayed import of Numpy when        | Replaced by build dependencies      |
+| building C extensions               | and isolation in `PEP 517`_ and     |
++-------------------------------------+ `PEP 518`_                          |
+| Transpilation of Cython to C before |                                     |
+| sdist                               |                                     |
++-------------------------------------+                                     |
+| Package specific versions of        |                                     |
+| astropy-helpers provided by git     |                                     |
+| submodule                           |                                     |
++-------------------------------------+-------------------------------------+
+
+We propose that a new package called extension-helpers be created starting from
```

Comment:
I am not convinced yet another package is necessary. Instead, we can show them how things should be done "natively" over at template?

Reply:
I don't quite understand how this would work as @pllim is proposing.  Then wouldn't we have to push updates to *all* users of this stuff if e.g. we need to update the `OpenMP` stuff?

Reply:
Our draft packaging guide shows both how to define extensions natively for simple cases and with extension-helpers:

https://oa-packaging-guide-preview.readthedocs.io/en/latest/extensions.html

For astropy, it's definitely the case that we have enough extensions that using extension-helpers would be a lot simpler than defining everything in the setup.py file, and the auto-detection of cython files is also a nice feature.

@pllim - are you satisfied with this or do you have specific wording recommendations for the APE?

Reply:
I think for packages that have one or two C extensions might not need to use `extension_helpers` as the standard ways of registering extensions would be sufficient. Certainly for core I think `extensions_helpers` would simplify things enough to warrant it's existence.

Reply:
Okay. I think I understand now. Thanks!